src/compiler/tracing.ts

```diff
@@ -122,16 +138,23 @@ namespace ts.tracing {
     }
     export function pop() {
         if (!traceFd) return;
-        Debug.assert(completeEvents.length > 0);
+        Debug.assert(canPop());

```

Comment:
Isn't it better to keep it at a low level of writing the expression directly to avoid the potential extra cost of a function call?

Reply:
While I'm almost certain it will be inlined, sure, I can duplicate the condition to speed up the assert.